src/stackmap_checker/call_stack_state.h

```diff
@@ -0,0 +1,63 @@
+#ifndef CALL_STACK_STATE_H
+#define CALL_STACK_STATE_H
+
+#include "stmap.h"
+#define UNW_LOCAL_ONLY
+#include <libunwind.h>
+
+#define MAX_CALL_STACK_DEPTH 256
+
+// The state of the call stack.
+typedef struct CallStackState {
+    // The address of the return address of each frame.
+    uint64_t *ret_addrs;
+    // The base pointers.
+    unw_word_t *bps;
+    // The 16 registers recorded for each frame.
+    unw_word_t **registers;
+    // The stack map records which correspond to each call on the stack.
+    stack_map_record_t *records;
+    // The number of frames on the stack.
+    uint32_t depth;
+} call_stack_state_t;
+
+/*
+ * Return the state of the call stack.
+ */
+call_stack_state_t* get_call_stack_state(unw_cursor_t cursor,
+                                         unw_context_t context);
+
+void free_call_stack_state(call_stack_state_t *state);
+
+/*
```

Comment:
Ah I see, you have documented what the functions do in the headers. I can roll with that.